doc/adrs/161_add_support_for_undo_redo_on_semantic_changes_in_representations.adoc

```diff
@@ -0,0 +1,185 @@
+= [ADR-161] Add support for undo redo on semantic changes in representations
+
+== Context
+
+We want to be able to undo or redo an action performed.
+
+
+=== Current behavior
+
+None.
+
+
+== Decision
+
+=== Frontend
+
+We will store the `ids` of each `mutation` performed by the front-end using `ApolloLink API` in two arrays to track available undo or redo.
+
+[source,typescript]
+----
+export class OperationCountLink extends ApolloLink {
+  constructor() {
+    super();
+  }
+  override request(operation: Operation, forward) {
+    if (
+      operation.query.definitions[0].kind === Kind.OPERATION_DEFINITION &&
+      operation.query.definitions[0].operation === OperationTypeNode.MUTATION &&
+      operation.variables.input.id &&
+      !(
+        operation.operationName === 'undo' ||
+        operation.operationName === 'redo'
+      )
+    ) {
+      var storedUndoStack = sessionStorage.getItem('undoStack');
+      var undoStack = JSON.parse(storedUndoStack);
+
+      sessionStorage.setItem('undoStack', JSON.stringify([operation.variables.input.id, ...undoStack]));
+      sessionStorage.setItem('redoStack', JSON.stringify([]));
+    }
+
+    return forward(operation);
+  }
+}
+----
+
+We will add two `event handlers` to handle `ctrl + z` and `ctrl + y` to respectively `undo` or `redo` a mutation.
+Theses event handler will send an undo or redo mutation to the back-end with the `id of the mutation` previously responsible for a `semantic change`.
+
+[source,typescript]
+----
+const undoLastAction = () => {
+var storedArray = sessionStorage.getItem('undoStack');
+if (storedArray) {
+    var arr = JSON.parse(storedArray);
+    if (arr[0]) {
+    const input: GQLUndoRedoInput = {
```

Comment:
Why an `UndoRedoInput`? Are there two mutations or one?

Reply:
The input contains the same variable but there are 2 mutations. I can make 2 inputs so it's more explicit.